src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -0,0 +1,367 @@
+
+        // adjust tour steps according to permissions
+        tour.steps.forEach((step, index) => {
+            if (step.permission && !this.accountService.hasAnyAuthorityDirect(step.permission)) {
+                this.currentTourSteps.splice(index, 1);
+            }
+        });
+
+        this._currentTour = cloneDeep(tour);
+        this._currentTour.steps = this._currentTour.steps.filter(step => !step.skipStep);
+        this._currentTourStepIndex = 0;
+        this._setFirstAndLast();
+        this._guidedTourOrbShowingSubject.next(this._currentTour.useOrb);
+        if (this._currentTour.steps.length > 0 && (!this._currentTour.minimumScreenSize || window.innerWidth >= this._currentTour.minimumScreenSize)) {
+            if (!this._currentTour.useOrb) {
+                document.body.classList.add('tour-open');
+            }
+            const currentStep = this._currentTour.steps[this._currentTourStepIndex];
+            if (currentStep.action) {
+                currentStep.action();
+            }
+
+            if (this._checkSelectorValidity()) {
+                this._guidedTourCurrentStepSubject.next(this.getPreparedTourStep(this._currentTourStepIndex));
+            } else {
+                this.nextStep();
+            }
+        }
+    }
+
+    /* Start tour with orb */
+    public activateOrb(): void {
+        this._guidedTourOrbShowingSubject.next(false);
+        document.body.classList.add('tour-open');
+    }
+
+    /**
+     * Define first and last tour step based on amount of tour steps
+     * @private
+     */
+    private _setFirstAndLast(): void {
+        if (this._currentTour) {
+            this._onLastStep = this._currentTour.steps.length - 1 === this._currentTourStepIndex;
+            this._onFirstStep = this._currentTourStepIndex === 0;
+        }
+    }
+
+    /* Check if highlighted element is available */
+    private _checkSelectorValidity(): boolean {
+        if (this._currentTour) {
+            if (this._currentTour.steps[this._currentTourStepIndex].selector) {
+                const selectedElement = document.querySelector(this._currentTour.steps[this._currentTourStepIndex].selector!);
+                if (!selectedElement) {
+                    this.errorHandler.handleError(
+                        // If error handler is configured this should not block the browser.
+                        new Error(
+                            `Error finding selector ${this._currentTour.steps[this._currentTourStepIndex].selector} on step ${this._currentTourStepIndex + 1} during guided tour: ${
+                                this._currentTour.settingsId
+                            }`,
+                        ),
+                    );
+                    return false;
+                }
+            }
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     *  Is last tour step
+     */
+    public get onLastStep(): boolean {
```

Comment:
I don't like some of the wording that much. E.g. `onLastStep` gives the impression that it is an event and a property. (better `isOnLastStep`)